### DISCUSSION TITLE
kafka connect No authorization

### DISCUSSION THREAD

**Discussion by chanvoyage** [strimzi/strimzi-kafka-operator#5652] [Q&A] [answered]
i was enabled  authorization type:simple  ,During  kafka cluster  restart for apply config that  Kafka connect cluster  No authorization to topic 

1).Do you have any suggestion for fix kafka connect  authorization  topic  or any document  guide

2). About  kafk user after  enable  authorization type:simple  i found something confused .

first time i config authorization to user "A"  can not  read topic  "A" , it work !  After that i  config again to  user  "A" can read topic "A"  
and test again found   log error tel me  user "A" No authorization Topic "A"

 
![image](https://user-images.githubusercontent.com/39899497/135840376-d69877c4-5f23-4bd5-bf72-0f4d16b61319.png)

![image](https://user-images.githubusercontent.com/39899497/135842728-943b75db-31d9-4f58-a7b6-104ad0ffec04.png)

**Comment by scholzj** [comment, marked as answer]
The picture with the log is too small to read. Please stop including screenshots and paste the logs as a code / text. Our security exampels use authorizaion and should give you an idea what right Connect might need: https://github.com/strimzi/strimzi-kafka-operator/tree/main/examples

**Reply by chanvoyage** [reply]
After enabled authorization type:simple kafka cluster restart and kafka connect cluster try to connect kafka cluster  error like below 

[kafka-connect-logs.txt](https://github.com/strimzi/strimzi-kafka-operator/files/7284146/kafka-connect-logs.txt)
![image](https://user-images.githubusercontent.com/39899497/135987657-aa4f6a0f-8960-4a11-ba4a-a90850aa52ff.png)


![image](https://user-images.githubusercontent.com/39899497/135988292-075bc9bc-e21a-4d25-85a2-812e60dd6827.png)

**Reply by scholzj** [reply]
Did you configured the ACLs I as suggested above? Did you configured authentication etc.?

**Reply by chanvoyage** [reply]
OK. after i try follow up security exampels  but kafka connect still  remains the same or Did i miss something ?

![image](https://user-images.githubusercontent.com/39899497/136157299-a9cc3bb8-064d-4697-ba59-b77275d94f43.png)

[config.txt](https://github.com/strimzi/strimzi-kafka-operator/files/7291618/config.txt)
[logs-error.txt](https://github.com/strimzi/strimzi-kafka-operator/files/7291621/logs-error.txt)

**Reply by scholzj** [reply]
Have you check the Kafka logs what they say about the missing rights? Have you checked if the resources are ready?

**Reply by chanvoyage** [reply]
kafka-connect  already but kafka log found message  User:ANONYMOUS is Denied Operation

![image](https://user-images.githubusercontent.com/39899497/136159918-50bff90a-971b-4f21-a1f0-99ad23728c5b.png)

[kafka-log.txt](https://github.com/strimzi/strimzi-kafka-operator/files/7291718/kafka-log.txt)

**Reply by scholzj** [reply]
Well, you don't have authentication enabled in the broker then I guess.

**Reply by chanvoyage** [reply]
Did you mean this one ? 

![image](https://user-images.githubusercontent.com/39899497/136161364-47e36205-e076-4e65-af29-29a449f9da6b.png)


apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: kafka-cluster
  namespace: kafka
spec:
  kafka:
    version: 2.8.0
    replicas: 3
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: tls
        port: 9093
        type: internal
        tls: true
      - name: external
        port: 9094
        type: ingress
        tls: true
        configuration:
          bootstrap:
            host: kafka-bootstrap.chaiyapon.com
          brokers:
          - broker: 0
            host: kafka-broker-0.chaiyapon.com
          - broker: 1
            host: kafka-broker-1.chaiyapon.com
          - broker: 2
            host: kafka-broker-2.chaiyapon.com 
        authentication:
          type: tls
    authorization:
      type: simple 
    config:
      offsets.topic.replication.factor: 3
      transaction.state.log.replication.factor: 3
      transaction.state.log.min.isr: 2
      log.message.format.version: "2.8"
      inter.broker.protocol.version: "2.8"

**Reply by scholzj** [reply]
Can you please format it as code? Without the formatting it is not possible to read the YAML structure.

**Reply by chanvoyage** [reply]
Is is ok 
[kafka.txt](https://github.com/strimzi/strimzi-kafka-operator/files/7291892/kafka.txt)

**Reply by scholzj** [reply]
Well, if you learn how to include it in the connect as a code, that will be the best way. Because nobody will need to download the file etc. I think you have the authentication badly aligned and configured only for one listener. So that is something to look into.

**Reply by chanvoyage** [reply]
I didn't know before I'm sorry scholzi

i have successfully about enabled tls , kafka connect talk to kafka cluster by port 9093  and i need to config that 

thank you very much
